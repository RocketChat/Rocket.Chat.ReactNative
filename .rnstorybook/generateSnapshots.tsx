import React from 'react';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import preview from './preview';
import { mockedStore } from '../app/reducers/mockedStore';
import { initStore } from '../app/lib/store/auxStore';

initStore(mockedStore);

export function generateSnapshots(stories: any) {
	describe('Story Snapshots:', () => {
		const composedStories = composeStories(stories, {
			decorators: preview.decorators
		});
		Object.entries(composedStories).forEach(([name, story]) => {
			test(`${name} should match snapshot`, () => {
				const rendered = render(React.createElement(story as React.ComponentType));
				expect(rendered.toJSON()).toMatchSnapshot();
			});
		});
	});
}
