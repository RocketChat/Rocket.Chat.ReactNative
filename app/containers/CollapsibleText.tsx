import React, { useState } from 'react';
import { TextStyle, Text, StyleSheet } from 'react-native';

import sharedStyles from '../views/Styles';
import { useTheme } from '../theme';
import { previewFormatText } from './markdown/previewFormatText';
import I18n from '../i18n';

interface ICollapsibleText {
	msg?: string;
	testID?: string;
	style?: TextStyle[];
	linesToTruncate?: number;
}

const styles = StyleSheet.create({
	text: {
		fontSize: 16,
		...sharedStyles.textRegular,
		textAlignVertical: 'center'
	},
	textInfo: {
		fontSize: 14,
		...sharedStyles.textRegular
	}
});

const CollapsibleText = ({ msg, style = [], testID, linesToTruncate = 1 }: ICollapsibleText) => {
	const [truncatedText, setTruncatedText] = useState('');
	const [showTruncated, setShowTruncated] = useState(true);

	const { colors } = useTheme();

	if (!msg) {
		return null;
	}

	const m = previewFormatText(msg);

	if (truncatedText && showTruncated) {
		return (
			<Text accessibilityLabel={truncatedText} style={[styles.text, { color: colors.bodyText }, ...style]}>
				{`${truncatedText}... `}
				<Text onPress={() => setShowTruncated(false)} style={[styles.textInfo, { color: colors.fontInfo }]}>
					{I18n.t('Show_more')}
				</Text>
			</Text>
		);
	}

	return (
		<Text
			accessibilityLabel={m}
			style={[styles.text, { color: colors.bodyText, height: !showTruncated ? undefined : 0 }, ...style]}
			testID={testID || `markdown-preview-${m}`}
			onTextLayout={event => {
				// get all lines
				const { lines } = event.nativeEvent;
				if (lines.length > linesToTruncate) {
					const text = lines
						.splice(0, linesToTruncate)
						.map(line => line.text)
						.join('');
					// 12 is equal the
					const truncatedTextLengthWithShowMore = text.length - (4 + I18n.t('Show_more').length);
					const clippedText = text.slice(0, truncatedTextLengthWithShowMore);
					setTruncatedText(clippedText);
				} else {
					setShowTruncated(false);
				}
			}}
		>
			{m}
			{truncatedText ? (
				<Text onPress={() => setShowTruncated(true)} style={[styles.textInfo, { color: colors.fontInfo }]}>
					{` ${I18n.t('Show_less')}`}
				</Text>
			) : null}
		</Text>
	);
};

export default CollapsibleText;
