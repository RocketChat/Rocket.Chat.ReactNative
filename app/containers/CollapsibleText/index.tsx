import React, { useState } from 'react';
import { TextStyle, Text, StyleSheet } from 'react-native';

import sharedStyles from '../../views/Styles';
import { useTheme } from '../../theme';
import { previewFormatText } from '../markdown/previewFormatText';
import I18n from '../../i18n';

interface ICollapsibleText {
	msg?: string;
	style?: TextStyle[];
	linesToTruncate?: number;
}

const styles = StyleSheet.create({
	text: {
		fontSize: 16,
		...sharedStyles.textRegular,
		textAlignVertical: 'center'
	},
	textInfo: {
		fontSize: 14,
		...sharedStyles.textRegular
	}
});

const CollapsibleText = ({ msg, style = [], linesToTruncate = 1 }: ICollapsibleText) => {
	const [truncatedText, setTruncatedText] = useState('');
	const [showTruncated, setShowTruncated] = useState(true);

	const { colors } = useTheme();

	if (!msg) {
		return null;
	}

	const m = previewFormatText(msg);

	if (truncatedText && showTruncated) {
		return (
			<Text testID={`collapsible-text-truncated-${m}`}>
				<Text accessibilityLabel={truncatedText} style={[styles.text, { color: colors.bodyText }, ...style]}>
					{`${truncatedText}... `}
				</Text>
				<Text onPress={() => setShowTruncated(false)} style={[styles.textInfo, { color: colors.actionTintColor }]}>
					{I18n.t('Show_more')}
				</Text>
			</Text>
		);
	}

	return (
		<Text
			accessibilityLabel={m}
			style={[styles.text, { color: colors.bodyText, height: !showTruncated ? undefined : 0 }, ...style]}
			testID={`collapsible-text-${m}`}
			onTextLayout={event => {
				const { lines } = event.nativeEvent;
				if (lines.length > linesToTruncate) {
					const text = lines
						.splice(0, linesToTruncate)
						.map(line => line.text)
						.join('');
					const truncatedTextLengthWithShowMore = text.length - (4 + I18n.t('Show_more').length);
					const clippedText = text.slice(0, truncatedTextLengthWithShowMore);
					setTruncatedText(clippedText);
				} else {
					setShowTruncated(false);
				}
			}}
		>
			{m}
			{truncatedText ? (
				<Text
					testID='collapsible-text-show-less'
					onPress={() => setShowTruncated(true)}
					style={[styles.textInfo, { color: colors.actionTintColor }]}
				>
					{` ${I18n.t('Show_less')}`}
				</Text>
			) : null}
		</Text>
	);
};

export default CollapsibleText;
