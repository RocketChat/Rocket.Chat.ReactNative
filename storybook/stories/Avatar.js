import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import { themes } from '../../app/constants/colors';
import Avatar from '../../app/containers/Avatar/Avatar';
import Status from '../../app/containers/Status/Status';
import StoriesSeparator from './StoriesSeparator';
import sharedStyles from '../../app/views/Styles';

const styles = StyleSheet.create({
	status: {
		borderWidth: 4,
		bottom: -4,
		right: -4
	},
	custom: {
		padding: 16
	}
});

const server = 'https://open.rocket.chat';

const Separator = ({ title, theme }) => <StoriesSeparator title={title} theme={theme} />;
Separator.propTypes = {
	title: PropTypes.string,
	theme: PropTypes.string
};

const AvatarStories = ({ theme }) => (
	<ScrollView style={{ backgroundColor: themes[theme].backgroundColor }}>
		<Separator title='Avatar by text' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by roomId' theme={theme} />
		<Avatar
			type='p'
			rid='devWBbYr7inwupPqK'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by url' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by path' theme={theme} />
		<Avatar
			avatar='/avatar/diego.mello'
			server={server}
			size={56}
		/>
		<Separator title='With ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			avatarETag='5ag8KffJcZj9m5rCv'
			server={server}
			size={56}
		/>
		<Separator title='Without ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			server={server}
			size={56}
		/>
		<Separator title='Emoji' theme={theme} />
		<Avatar
			emoji='troll'
			getCustomEmoji={() => ({ name: 'troll', extension: 'jpg' })}
			server={server}
			size={56}
		/>
		<Separator title='Direct' theme={theme} />
		<Avatar
			text='diego.mello'
			server={server}
			type='d'
			size={56}
		/>
		<Separator title='Channel' theme={theme} />
		<Avatar
			text='general'
			server={server}
			type='c'
			size={56}
		/>
		<Separator title='Touchable' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			onPress={() => console.log('Pressed!')}
			size={56}
		/>
		<Separator title='Static' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			isStatic
			size={56}
		/>
		<Separator title='Custom borderRadius' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			borderRadius={28}
			size={56}
		/>
		<Separator title='Children' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		>
			<Status
				size={24}
				style={[sharedStyles.status, styles.status]}
				theme={theme}
			/>
		</Avatar>
		<Separator title='Wrong server' theme={theme} />
		<Avatar
			text='Avatar'
			server='https://google.com'
			size={56}
		/>
		<Separator title='Custom style' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
			style={styles.custom}
		/>
	</ScrollView>
);
AvatarStories.propTypes = {
	theme: PropTypes.string
};
export default AvatarStories;
