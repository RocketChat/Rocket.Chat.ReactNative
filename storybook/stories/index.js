/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { storiesOf } from '@storybook/react-native';

import RoomItem from './RoomItem';
import Message from './Message';
import UiKitMessage from './UiKitMessage';
import UiKitModal from './UiKitModal';
import Markdown from './Markdown';
// import RoomViewHeader from './RoomViewHeader';

import MessageContext from '../../app/containers/message/Context';

// MessageProvider
const baseUrl = 'https://open.rocket.chat';
const user = {
	id: '',
	username: 'diego.mello',
	token: ''
};

// Change here to see themed storybook
const theme = 'light';

const reducers = combineReducers({
	settings: () => ({}),
	login: () => ({
		user: {
			username: 'diego.mello'
		}
	}),
	meteor: () => ({ connected: true }),
	activeUsers: () => ({ abc: { status: 'online', statusText: 'dog' } })
});
const store = createStore(reducers);

const messageDecorator = story => (
	<MessageContext.Provider
		value={{
			user,
			baseUrl,
			onPress: () => {},
			onLongPress: () => {},
			reactionInit: () => {},
			onErrorPress: () => {},
			replyBroadcast: () => {},
			onReactionPress: () => {},
			onDiscussionPress: () => {},
			onReactionLongPress: () => {}
		}}
	>
		{story()}
	</MessageContext.Provider>
);

storiesOf('RoomItem', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.add('list roomitem', () => <RoomItem theme={theme} />);
storiesOf('Message', module)
	.addDecorator(messageDecorator)
	.add('list message', () => <Message theme={theme} />);

storiesOf('UiKitMessage', module)
	.addDecorator(messageDecorator)
	.add('list uikitmessage', () => <UiKitMessage theme={theme} />);
storiesOf('UiKitModal', module)
	.addDecorator(messageDecorator)
	.add('list UiKitModal', () => <UiKitModal theme={theme} />);
storiesOf('Markdown', module)
	.add('list Markdown', () => <Markdown theme={theme} />);

// FIXME: I couldn't make these pass on jest :(
// storiesOf('RoomViewHeader', module)
// 	.add('list', () => <RoomViewHeader theme='black' />);
