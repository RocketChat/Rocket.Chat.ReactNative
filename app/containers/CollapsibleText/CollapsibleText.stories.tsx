import React from 'react';
import { View } from 'react-native';

import CollapsibleText from '.';

const smallText = 'Lorem ipsum dolor sit amet';

const text120 =
	'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vel vestibulum neque. Proin dignissim neque in urna nec.';

export default {
	title: 'Collapsible Text'
};

export const Item = () => (
	<View style={{ padding: 20 }}>
		<CollapsibleText linesToTruncate={1} msg={`${smallText}`} />
		<CollapsibleText linesToTruncate={1} msg={`linesToTruncate: 1 - ${text120}`} />
		<CollapsibleText linesToTruncate={2} msg={`linesToTruncate: 2 - ${text120}`} />
	</View>
);
