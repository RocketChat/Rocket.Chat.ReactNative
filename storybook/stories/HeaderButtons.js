/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions, react/prop-types */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import * as HeaderButton from '../../app/containers/HeaderButton';
import Header from '../../app/containers/Header';
import { ThemeContext } from '../../app/theme';

const stories = storiesOf('Header Buttons', module);

const HeaderExample = ({ left, right }) => (
	<Header
		headerLeft={left}
		headerTitle={() => <View style={{ flex: 1 }} />}
		headerRight={right}
	/>
);

stories.add('title', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('icons', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('badge', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadUser={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadGroup={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</>
));

const ThemeStory = ({ theme }) => (
	<ThemeContext.Provider
		value={{ theme }}
	>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='Threads' />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</ThemeContext.Provider>
);

stories.add('themes', () => (
	<>
		<ThemeStory theme='light' />
		<ThemeStory theme='dark' />
		<ThemeStory theme='black' />
	</>
));

stories.add('common', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Drawer />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CloseModal />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CancelModal />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.More />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Download />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Preferences />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Legal />
			)}
		/>
	</>
));
