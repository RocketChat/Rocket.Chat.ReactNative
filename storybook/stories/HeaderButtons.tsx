import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { SafeAreaView } from 'react-native';
import { Header, HeaderBackground } from '@react-navigation/elements';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import * as HeaderButton from '../../app/containers/HeaderButton';
import { TColors, ThemeContext, TSupportedThemes } from '../../app/theme';
import { colors } from '../../app/lib/constants';

const stories = storiesOf('Header Buttons', module).addDecorator(story => <SafeAreaProvider>{story()}</SafeAreaProvider>);

interface IHeader {
	left?: () => React.ReactElement | null;
	right?: () => React.ReactElement;
	title?: string;
	colors?: TColors;
}

const HeaderExample = ({ left, right, colors, title = '' }: IHeader) => (
	<SafeAreaView>
		<Header
			title={title}
			headerLeft={left}
			headerRight={right}
			headerBackground={() => <HeaderBackground style={{ backgroundColor: colors?.headerBackground }} />}
		/>
	</SafeAreaView>
);

stories.add('title', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('icons', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('badge', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadUser={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadGroup={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</>
));

const ThemeStory = ({ theme }: { theme: TSupportedThemes }) => (
	<ThemeContext.Provider value={{ theme, colors: colors[theme] }}>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='Threads' />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
				</HeaderButton.Container>
			)}
			colors={colors[theme]}
		/>
	</ThemeContext.Provider>
);

stories.add('themes', () => (
	<>
		<ThemeStory theme='light' />
		<ThemeStory theme='dark' />
		<ThemeStory theme='black' />
	</>
));

stories.add('common', () => (
	<>
		<HeaderExample left={() => <HeaderButton.Drawer />} />
		<HeaderExample left={() => <HeaderButton.CloseModal />} />
		<HeaderExample left={() => <HeaderButton.CancelModal />} />
		<HeaderExample right={() => <HeaderButton.More />} />
		<HeaderExample right={() => <HeaderButton.Download />} />
		<HeaderExample right={() => <HeaderButton.Preferences />} />
		<HeaderExample right={() => <HeaderButton.Legal />} />
	</>
));
