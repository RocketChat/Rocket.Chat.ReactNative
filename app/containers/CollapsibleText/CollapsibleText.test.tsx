import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './CollapsibleText.stories';

generateSnapshots(stories);
