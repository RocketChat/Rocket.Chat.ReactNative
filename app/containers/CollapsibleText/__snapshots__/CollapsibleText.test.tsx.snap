// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: Item should match snapshot 1`] = `
<View
  style={
    {
      "padding": 20,
    }
  }
>
  <Text
    accessibilityLabel="Lorem ipsum dolor sit amet"
    onTextLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "transparent",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "400",
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#2F343D",
          "height": 0,
        },
      ]
    }
    testID="collapsible-text-Lorem ipsum dolor sit amet"
  >
    Lorem ipsum dolor sit amet
  </Text>
  <Text
    accessibilityLabel="linesToTruncate: 1 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vel vestibulum neque. Proin dignissim neque in urna nec."
    onTextLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "transparent",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "400",
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#2F343D",
          "height": 0,
        },
      ]
    }
    testID="collapsible-text-linesToTruncate: 1 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vel vestibulum neque. Proin dignissim neque in urna nec."
  >
    linesToTruncate: 1 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vel vestibulum neque. Proin dignissim neque in urna nec.
  </Text>
  <Text
    accessibilityLabel="linesToTruncate: 2 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vel vestibulum neque. Proin dignissim neque in urna nec."
    onTextLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "transparent",
          "fontFamily": "Inter",
          "fontSize": 16,
          "fontWeight": "400",
          "textAlign": "left",
          "textAlignVertical": "center",
        },
        {
          "color": "#2F343D",
          "height": 0,
        },
      ]
    }
    testID="collapsible-text-linesToTruncate: 2 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vel vestibulum neque. Proin dignissim neque in urna nec."
  >
    linesToTruncate: 2 - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vel vestibulum neque. Proin dignissim neque in urna nec.
  </Text>
</View>
`;
